File: BMDInfo-server/src/main/java/com/bd/spectrum/BMDInfo_server/model/BidTracker.java
package com.bd.spectrum.BMDInfo_server.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class BidTracker {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String client;
    @Column(name = "description", length = 1000)
    private String description;
    private String cam;
    private String salesResource;
    private String bidManager;
    private String initiationMode;
    private String stage;
    private String procurementType;
    private String goods;
    private String works;
    private String service;
    private String tenderId;
    private String initiation;
    private String published_initiation;
    @JsonFormat(pattern = "EEE, MMM dd, yyyy", locale = "en")
    private LocalDate published;
    @JsonFormat(pattern = "EEE, MMM dd, yyyy", locale = "en")
    private LocalDate preBidDate;
    @JsonFormat(pattern = "h:mm:ss a", locale = "en")
    private LocalTime preBidTime;
    @JsonFormat(pattern = "EEE, MMM dd, yyyy", locale = "en")
    private LocalDate submissionDate;
    @JsonFormat(pattern = "h:mm:ss a", locale = "en")
    private LocalTime submissionTime;
    private String securityMode;
    private BigDecimal securityAmount;
    private BigDecimal creditFacility;
    @JsonFormat(pattern = "EEE, MMM dd, yyyy", locale = "en")
    private LocalDate issuingDate;
    private String referenceNumber;
    private String issuingBank;
    @JsonFormat(pattern = "dd MMM,yyyy", locale = "en")
    private LocalDate expiryDate;
    private String submission;
    private String result;
    private String noName;
    @Column(name = "remarks", length = 1000)
    private String remarks;
}

File: BMDInfo-client/src/app/interface/bid-tracker.ts
export interface BidTracker {
  id?: number; // optional if not present during creation

  client: string;
  description: string;
  cam: string;
  salesResource: string;
  bidManager: string;
  initiationMode: string;
  stage: string;
  procurementType: string;
  goods: string;
  works: string;
  service: string;
  tenderId: string;
  initiation: string;
  published_initiation: string;

  published: string;         // Format: "EEE, MMM dd, yyyy"
  preBidDate: string;        // Format: "EEE, MMM dd, yyyy"
  preBidTime: string;        // Format: "h:mm:ss a"
  submissionDate: string;    // Format: "EEE, MMM dd, yyyy"
  submissionTime: string;    // Format: "h:mm:ss a"

  securityMode: string;
  securityAmount: number;
  creditFacility: number;
  issuingDate: string;       // Format: "EEE, MMM dd, yyyy"

  referenceNumber: string;
  issuingBank: string;
  expiryDate: string;              // Format: "dd MMM,yyyy"
  submission: string;
  result: string;
  noName: string;
  remarks: string;
}
